sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -0,0 +1,711 @@
+    return new AutoValue_RabbitMqIO_Read.Builder()
+        .setConnectionConfig(ConnectionConfig.create()).setQueueDeclare(false)
+        .setMaxReadTime(null).setMaxNumRecords(Long.MAX_VALUE).setUseCorrelationId(false).build();
+  }
+
+  public static Write write() {
+    return new AutoValue_RabbitMqIO_Write.Builder()
+        .setConnectionConfig(ConnectionConfig.create())
+        .setExchangeDeclare(false).build();
+  }
+
+  private RabbitMqIO() {
+  }
+
+  /**
+   * Describe a connection configuration to a RabbitMQ server.
+   */
+  @AutoValue
+  public abstract static class ConnectionConfig implements Serializable {
+
+    @Nullable abstract String uri();
+
+    abstract int networkRecoveryInterval();
+    abstract boolean automaticRecovery();
+    abstract boolean topologyRecovery();
+
+    abstract int connectionTimeout();
+    abstract int requestedChannelMax();
+    abstract int requestedFrameMax();
+    abstract int requestedHeartbeat();
+
+    abstract Builder builder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setUri(String uri);
+      abstract Builder setNetworkRecoveryInterval(int networkRecoveryInterval);
+      abstract Builder setAutomaticRecovery(boolean automaticRecovery);
+      abstract Builder setTopologyRecovery(boolean topologyRecovery);
+      abstract Builder setConnectionTimeout(int connectionTimeout);
+      abstract Builder setRequestedChannelMax(int requestedChannelMax);
+      abstract Builder setRequestedFrameMax(int requestedFrameMax);
+      abstract Builder setRequestedHeartbeat(int requestedHeartbeat);
+      abstract ConnectionConfig build();
+    }
+
+    public static ConnectionConfig create() {
+      return new AutoValue_RabbitMqIO_ConnectionConfig.Builder()
+          .setUri("amqp://localhost:5672")
+          .setAutomaticRecovery(true)
+          .setTopologyRecovery(true)
+          .setConnectionTimeout(60000)
+          .setRequestedChannelMax(0)
+          .setRequestedFrameMax(0)
+          .setRequestedHeartbeat(60)
+          .setNetworkRecoveryInterval(5000)
+          .build();
+    }
+
+    /**
+     * Create a RabbitMQ connection configuration with broker URI and a queue name.
+     *
+     * @param uri The RabbitMQ server URI.
+     * @return The corresponding {@link ConnectionConfig}.
+     */
+    public static ConnectionConfig create(String uri) {
+      checkArgument(uri != null, "uri can not be null");
+      return new AutoValue_RabbitMqIO_ConnectionConfig.Builder()
+          .setUri(uri)
+          .setAutomaticRecovery(true)
+          .setTopologyRecovery(true)
+          .setConnectionTimeout(60000)
+          .setRequestedChannelMax(0)
+          .setRequestedFrameMax(0)
+          .setRequestedHeartbeat(60)
+          .setNetworkRecoveryInterval(5000)
+          .build();
+    }
+
+    /**
+     * Define the RabbitMQ URI.
+     *
+     * @param uri The RabbitMQ URI.
+     * @return The corresponding {@link ConnectionConfig}.
```

Comment:
These javadocs are not useful. This also keeps coming up in almost every IO review - please keep in mind for future IOs.